### ISSUE TITLE
PyPy release page moved from bitbucket to python.org

### ISSUE THREAD

**Issue by ashwinvis** [pyenv/pyenv#1681]
Too many issues will kill our team's development velocity, drastically.
Make sure you have checked all steps below.

### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [x] Make sure you are not asking us to help solving your specific issue.
  * GitHub issues is opened mainly for development purposes. If you want to ask someone to help solving your problem, go to some community site like [Gitter](https://gitter.im/yyuu/pyenv), [StackOverflow](https://stackoverflow.com/questions/tagged/pyenv), etc.
* [x] Make sure your problem is not derived from packaging (e.g. [Homebrew](https://brew.sh)).
  * Please refer to the package documentation for the installation issues, etc.
* [x] Make sure your problem is not derived from plugins.
  * This repository is maintaining `pyenv` and the default `python-build` plugin only. Please refrain from reporting issues of other plugins here.

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): Arch Linux
- [x] OS architecture (e.g. amd64): amd64
- [x] pyenv version: 1.2.20-1-g70635dad
- [x] Python version: 3.8.1
- [x] C Compiler information (e.g. gcc 7.3):  gcc 10.1.0
- [x] Please attach verbose build log as gist
  * You can turn on verbose debug logging using by setting `PYENV_DEBUG=1`, e.g. `env PYENV_DEBUG=1 pyenv install -v 3.6.4`

```sh
❯ PYENV_DEBUG=1 pyenv install -v pypy3.6-7.3.1                                                                                                                                              
+ [pyenv:22] enable -f /home/avmo/.pyenv/bin/../libexec/pyenv-realpath.dylib realpath
+ [pyenv:29] '[' -z '' ']'
++ [pyenv:31] type -p greadlink readlink
++ [pyenv:31] head -1
+ [pyenv:31] READLINK=/usr/bin/readlink
+ [pyenv:32] '[' -n /usr/bin/readlink ']'
+ [pyenv:57] '[' -z '' ']'
+ [pyenv:58] PYENV_ROOT=/home/avmo/.pyenv
+ [pyenv:62] export PYENV_ROOT
+ [pyenv:65] '[' -z '' ']'
+ [pyenv:66] '[' -n '' ']'
+ [pyenv:82] '[' -z '' ']'
+ [pyenv:83] PYENV_DIR=/home/avmo
+ [pyenv:86] '[' '!' -d /home/avmo ']'
+ [pyenv:86] '[' '!' -e /home/avmo ']'
++ [pyenv:90] cd /home/avmo
++ [pyenv:90] echo /home/avmo
+ [pyenv:90] PYENV_DIR=/home/avmo
+ [pyenv:91] export PYENV_DIR
+ [pyenv:94] shopt -s nullglob
++ [pyenv:96] abs_dirname /home/avmo/.pyenv/bin/pyenv
++ [pyenv:39] local path=/home/avmo/.pyenv/bin/pyenv
++ [pyenv:43] '[' -n /home/avmo/.pyenv/bin/pyenv ']'
++ [pyenv:44] cd_path=/home/avmo/.pyenv/bin
++ [pyenv:45] [[ /home/avmo/.pyenv/bin != \/\h\o\m\e\/\a\v\m\o\/\.\p\y\e\n\v\/\b\i\n\/\p\y\e\n\v ]]
++ [pyenv:46] cd /home/avmo/.pyenv/bin
++ [pyenv:48] name=pyenv
+++ [pyenv:49] resolve_link pyenv
+++ [pyenv:35] /usr/bin/readlink pyenv
++ [pyenv:49] path=../libexec/pyenv
++ [pyenv:43] '[' -n ../libexec/pyenv ']'
++ [pyenv:44] cd_path=../libexec
++ [pyenv:45] [[ ../libexec != \.\.\/\l\i\b\e\x\e\c\/\p\y\e\n\v ]]
++ [pyenv:46] cd ../libexec
++ [pyenv:48] name=pyenv
+++ [pyenv:49] resolve_link pyenv
+++ [pyenv:35] /usr/bin/readlink pyenv
+++ [pyenv:49] true
++ [pyenv:49] path=
++ [pyenv:43] '[' -n '' ']'
++ [pyenv:52] echo /home/avmo/.pyenv/libexec
+ [pyenv:96] bin_path=/home/avmo/.pyenv/libexec
+ [pyenv:97] for plugin_bin in "${PYENV_ROOT}/plugins/"*/bin
+ [pyenv:98] PATH=/home/avmo/.pyenv/plugins/python-build/bin:/home/avmo/.local/bin/conda-app:/home/avmo/.cargo/bin:/home/avmo/.go/bin:/home/avmo/.npm-packages/bin:/home/avmo/.local/share/perl5/bin:/opt/miniconda3/condabin:/home/avmo/.pyenv/shims:/home/avmo/.pyenv/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/opt/cuda/bin:/home/avmo/.local/share/flatpak/exports/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/avmo/.local/bin:/home/avmo/.gem/ruby/2.7.0/bin:/home/avmo/.yarn/bin:
+ [pyenv:100] export PATH=/home/avmo/.pyenv/libexec:/home/avmo/.pyenv/plugins/python-build/bin:/home/avmo/.local/bin/conda-app:/home/avmo/.cargo/bin:/home/avmo/.go/bin:/home/avmo/.npm-packages/bin:/home/avmo/.local/share/perl5/bin:/opt/miniconda3/condabin:/home/avmo/.pyenv/shims:/home/avmo/.pyenv/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/opt/cuda/bin:/home/avmo/.local/share/flatpak/exports/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/avmo/.local/bin:/home/avmo/.gem/ruby/2.7.0/bin:/home/avmo/.yarn/bin:
+ [pyenv:100] PATH=/home/avmo/.pyenv/libexec:/home/avmo/.pyenv/plugins/python-build/bin:/home/avmo/.local/bin/conda-app:/home/avmo/.cargo/bin:/home/avmo/.go/bin:/home/avmo/.npm-packages/bin:/home/avmo/.local/share/perl5/bin:/opt/miniconda3/condabin:/home/avmo/.pyenv/shims:/home/avmo/.pyenv/bin:/usr/local/bin:/usr/local/sbin:/usr/bin:/opt/cuda/bin:/home/avmo/.local/share/flatpak/exports/bin:/var/lib/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/avmo/.local/bin:/home/avmo/.gem/ruby/2.7.0/bin:/home/avmo/.yarn/bin:
+ [pyenv:102] PYENV_HOOK_PATH=:/home/avmo/.pyenv/pyenv.d
+ [pyenv:103] '[' /home/avmo/.pyenv '!=' /home/avmo/.pyenv ']'
+ [pyenv:107] PYENV_HOOK_PATH=:/home/avmo/.pyenv/pyenv.d:/usr/local/etc/pyenv.d:/etc/pyenv.d:/usr/lib/pyenv/hooks
+ [pyenv:111] PYENV_HOOK_PATH=/home/avmo/.pyenv/pyenv.d:/usr/local/etc/pyenv.d:/etc/pyenv.d:/usr/lib/pyenv/hooks
+ [pyenv:112] export PYENV_HOOK_PATH
+ [pyenv:114] shopt -u nullglob
+ [pyenv:117] command=install
+ [pyenv:118] case "$command" in
++ [pyenv:131] command -v pyenv-install
+ [pyenv:131] command_path=/home/avmo/.pyenv/plugins/python-build/bin/pyenv-install
+ [pyenv:132] '[' -z /home/avmo/.pyenv/plugins/python-build/bin/pyenv-install ']'
+ [pyenv:140] shift 1
+ [pyenv:141] '[' -v = --help ']'
+ [pyenv:148] exec /home/avmo/.pyenv/plugins/python-build/bin/pyenv-install -v pypy3.6-7.3.1
+ [pyenv-install:32] shopt -s nullglob
+ [pyenv-install:33] for plugin_path in "$PYENV_ROOT"/plugins/*/share/python-build
+ [pyenv-install:34] PYTHON_BUILD_DEFINITIONS=:/home/avmo/.pyenv/plugins/python-build/share/python-build
+ [pyenv-install:36] export PYTHON_BUILD_DEFINITIONS
+ [pyenv-install:37] shopt -u nullglob
+ [pyenv-install:40] '[' -v = --complete ']'
++ [pyenv-install:53] python-build --lib
+ [pyenv-install:53] eval 'lib () 
{ 
    function parse_options () 
    { 
        OPTIONS=();
        ARGUMENTS=();
        local arg option index;
        for arg in "$@";
        do
            if [ "${arg:0:1}" = "-" ]; then
                if [ "${arg:1:1}" = "-" ]; then
                    OPTIONS[${#OPTIONS[*]}]="${arg:2}";
                else
                    index=1;
                    while option="${arg:$index:1}"; do
                        [ -n "$option" ] || break;
                        OPTIONS[${#OPTIONS[*]}]="$option";
                        index=$(($index+1));
                    done;
                fi;
            else
                ARGUMENTS[${#ARGUMENTS[*]}]="$arg";
            fi;
        done
    };
    if [ "$1" == "--$FUNCNAME" ]; then
        declare -f "$FUNCNAME";
        echo "$FUNCNAME \"\$1\";";
        exit;
    fi
}
lib "$1";'
++ [pyenv-install:84] lib -v
++ [pyenv-install:78] '[' -v == --lib ']'
+ [pyenv-install:69] unset FORCE
+ [pyenv-install:70] unset SKIP_EXISTING
+ [pyenv-install:71] unset KEEP
+ [pyenv-install:72] unset VERBOSE
+ [pyenv-install:73] unset HAS_PATCH
+ [pyenv-install:74] unset DEBUG
+ [pyenv-install:76] parse_options -v pypy3.6-7.3.1
+ [pyenv-install:57] OPTIONS=()
+ [pyenv-install:58] ARGUMENTS=()
+ [pyenv-install:59] local arg option index
+ [pyenv-install:60] for arg in "$@"
+ [pyenv-install:62] '[' - = - ']'
+ [pyenv-install:63] '[' v = - ']'
+ [pyenv-install:66] index=1
+ [pyenv-install:67] option=v
+ [pyenv-install:68] '[' -n v ']'
+ [pyenv-install:69] OPTIONS[${#OPTIONS[*]}]=v
+ [pyenv-install:70] index=2
+ [pyenv-install:67] option=
+ [pyenv-install:68] '[' -n '' ']'
+ [pyenv-install:68] break
+ [pyenv-install:60] for arg in "$@"
+ [pyenv-install:62] '[' p = - ']'
+ [pyenv-install:74] ARGUMENTS[${#ARGUMENTS[*]}]=pypy3.6-7.3.1
+ [pyenv-install:77] for option in "${OPTIONS[@]}"
+ [pyenv-install:78] case "$option" in
+ [pyenv-install:97] VERBOSE=-v
+ [pyenv-install:114] '[' 1 -le 1 ']'
+ [pyenv-install:116] unset VERSION_NAME
+ [pyenv-install:122] DEFINITION=pypy3.6-7.3.1
+ [pyenv-install:123] '[' -n pypy3.6-7.3.1 ']'
+ [pyenv-install:124] '[' -n pypy3.6-7.3.1 ']'
+ [pyenv-install:129] declare -a before_hooks after_hooks
+ [pyenv-install:141] OLDIFS=' 
'
+ [pyenv-install:142] IFS='
'
+ [pyenv-install:142] scripts=(`pyenv-hooks install`)
++ [pyenv-install:142] pyenv-hooks install
+ [pyenv-hooks:9] '[' install = --complete ']'
+ [pyenv-hooks:18] PYENV_COMMAND=install
+ [pyenv-hooks:19] '[' -z install ']'
+ [pyenv-hooks:24] enable -f /home/avmo/.pyenv/libexec/pyenv-realpath.dylib realpath
+ [pyenv-hooks:25] '[' -n '' ']'
++ [pyenv-hooks:29] type -p greadlink readlink
++ [pyenv-hooks:29] head -1
+ [pyenv-hooks:29] READLINK=/usr/bin/readlink
+ [pyenv-hooks:30] '[' -z /usr/bin/readlink ']'
+ [pyenv-hooks:55] IFS=:
+ [pyenv-hooks:55] hook_paths=($PYENV_HOOK_PATH)
+ [pyenv-hooks:57] shopt -s nullglob
+ [pyenv-hooks:58] for path in "${hook_paths[@]}"
+ [pyenv-hooks:58] for path in "${hook_paths[@]}"
+ [pyenv-hooks:58] for path in "${hook_paths[@]}"
+ [pyenv-hooks:58] for path in "${hook_paths[@]}"
+ [pyenv-hooks:63] shopt -u nullglob
+ [pyenv-install:143] IFS=' 
'
+ [pyenv-install:149] '[' -n '' ']'
+ [pyenv-install:149] VERSION_NAME=pypy3.6-7.3.1
+ [pyenv-install:150] '[' -n '' ']'
+ [pyenv-install:151] PREFIX=/home/avmo/.pyenv/versions/pypy3.6-7.3.1
+ [pyenv-install:153] '[' -d /home/avmo/.pyenv/versions/pypy3.6-7.3.1 ']'
+ [pyenv-install:157] '[' -d /home/avmo/.pyenv/versions/pypy3.6-7.3.1/bin ']'
+ [pyenv-install:175] '[' -n '' ']'
+ [pyenv-install:182] '[' -z '' ']'
+ [pyenv-install:182] '[' -d /home/avmo/.pyenv/cache ']'
+ [pyenv-install:183] export PYTHON_BUILD_CACHE_PATH=/home/avmo/.pyenv/cache
+ [pyenv-install:183] PYTHON_BUILD_CACHE_PATH=/home/avmo/.pyenv/cache
+ [pyenv-install:186] '[' -z '' ']'
+ [pyenv-install:187] case "${VERSION_NAME}" in
+ [pyenv-install:230] '[' -n '' ']'
+ [pyenv-install:242] trap cleanup SIGINT
+ [pyenv-install:245] STATUS=0
+ [pyenv-install:246] python-build -v pypy3.6-7.3.1 /home/avmo/.pyenv/versions/pypy3.6-7.3.1
/tmp/python-build.20200819113541.964904 ~
Downloading pypy3.6-v7.3.1-linux64.tar.bz2...

08/19 11:35:41 [NOTICE] Downloading 1 item(s)

08/19 11:35:41 [ERROR] CUID#7 - Download aborted. URI=https://pyenv.github.io/pythons/f67cf1664a336a3e939b58b3cabfe47d893356bdc01f2e17bc912aaa6605db12
Exception: [AbstractCommand.cc:351] errorCode=3 URI=https://pyenv.github.io/pythons/f67cf1664a336a3e939b58b3cabfe47d893356bdc01f2e17bc912aaa6605db12
  -> [HttpSkipResponseCommand.cc:218] errorCode=3 Resource not found

08/19 11:35:41 [NOTICE] Download GID#e51d4570f7587b9e not complete: 

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
e51d45|ERR |       0B/s|https://pyenv.github.io/pythons/f67cf1664a336a3e939b58b3cabfe47d893356bdc01f2e17bc912aaa6605db12

Status Legend:
(ERR):error occurred.

aria2 will resume download if the transfer is restarted.
If there are any errors, then see the log file. See '-l' option in help/man page for details.
-> https://bitbucket.org/pypy/pypy/downloads/pypy3.6-v7.3.1-linux64.tar.bz2

08/19 11:35:41 [NOTICE] Downloading 1 item(s)

08/19 11:35:41 [ERROR] CUID#7 - Download aborted. URI=https://bitbucket.org/pypy/pypy/downloads/pypy3.6-v7.3.1-linux64.tar.bz2
Exception: [AbstractCommand.cc:351] errorCode=22 URI=https://bitbucket.org/pypy/pypy/downloads/pypy3.6-v7.3.1-linux64.tar.bz2
  -> [HttpSkipResponseCommand.cc:239] errorCode=22 The response status is not successful. status=403

08/19 11:35:41 [NOTICE] Download GID#2fdf0cf07c9ed272 not complete: /tmp/python-build.20200819113541.964904/pypy3.6-v7.3.1-linux64.tar.bz2

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
2fdf0c|ERR |       0B/s|/tmp/python-build.20200819113541.964904/pypy3.6-v7.3.1-linux64.tar.bz2

Status Legend:
(ERR):error occurred.

aria2 will resume download if the transfer is restarted.
If there are any errors, then see the log file. See '-l' option in help/man page for details.
error: failed to download pypy3.6-v7.3.1-linux64.tar.bz2

BUILD FAILED (Arch rolling using python-build 1.2.20-1-g70635dad)

+ [pyenv-install:246] STATUS=1
+ [pyenv-install:249] '[' 1 == 2 ']'
+ [pyenv-install:277] '[' 1 == 0 ']'
+ [pyenv-install:280] cleanup
+ [pyenv-install:239] '[' -z '' ']'
+ [pyenv-install:239] rm -rf /home/avmo/.pyenv/versions/pypy3.6-7.3.1
+ [pyenv-install:283] exit 1

```

See also:
* https://foss.heptapod.net/pypy/pypy.org/-/merge_requests/18/diffs
* https://downloads.python.org/pypy/

**Comment by ashwinvis** [comment]
I am trying to carefully replace the links with an awk script:
```awk
#!/bin/awk -f

/install_.*bitbucket.org.pypy/{
  new=$3
  gsub(/bitbucket.org\/pypy\/pypy\/downloads/, "downloads.python.org/pypy", new);

  cmd = "curl -s -o /dev/null -i -w '%{http_code}' "new
  cmd | getline result
  close(cmd)

  if (result == "200") {
    # print "Working"
    $3=new
  } else {
    # print "Not Working"
  }
  print
}

!/install_.*bitbucket.org.pypy/{print}
```

If it works, I can submit a PR.

Any idea what to do with portable-pypy links? It moved to GitHub but only v7.2.0 is available. https://github.com/squeaky-pl/portable-pypy/releases/

**Comment by squeaky-pl** [comment]
@ashwinvis Portable PyPy builds were archived here:  https://bitbucket-archive.softwareheritage.org/projects/sq/squeaky/portable-pypy.html

All the links to portable builds can be changed to start with https://bitbucket-archive.softwareheritage.org/static/14/140b7b14-aa94-424e-b191-9cd3438381f7/attachments/ and things should start working again.

**Comment by ashwinvis** [comment]
pypy-1.5 is not archived, so I propose to drop that. If needed, we can retain pypy-1.5-src pointing to the release on https://foss.heptapod.net/pypy/pypy/-/tags.

**Comment by ashwinvis** [comment]
Fixed in #1682
